### PR TITLE
[ISSUE #680] Fix the listTable in HadoopCatalog

### PR DESCRIPTION
**Pull request by shawnding**
Check the metadata directory is an iceberg table directory, use the file `.metadata.json` .

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java [hunk 1]
```diff
@@ -195,6 +198,8 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     throw new UnsupportedOperationException("Cannot rename Hadoop tables");
   }
 
+  private static final PathFilter TABLE_FILTER = path -> path.getName().endsWith(TABLE_METADATA_FILE_EXTENSION);
```
Comment (jerryshao) [inline core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java]: Should this class variable move to the top of the class?

Comment (shawnding) [reply core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java]: OK moved

Comment (jerryshao) [reply core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java]: I would suggest it to align with other `static final` variables.

Comment (shawnding) [reply core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java]: done

**Comment by jerryshao** [comment]
+1, LGTM.

**Comment by jerryshao** [comment]
@rdblue @aokolnychyi mind taking a look at this PR? Thanks.

File: core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java [hunk 2]
```diff
@@ -59,9 +60,12 @@
  */
 public class HadoopCatalog extends BaseMetastoreCatalog implements Closeable {
   private static final String ICEBERG_HADOOP_WAREHOUSE_BASE = "iceberg/warehouse";
+  private static final String TABLE_METADATA_FILE_EXTENSION = ".metadata.json";
+  private static final PathFilter TABLE_FILTER = path -> path.getName().endsWith(TABLE_METADATA_FILE_EXTENSION);
   private final Configuration conf;
   private String warehouseLocation;
 
+
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java]: Nit: Please remove additional blank lines like this.

**Comment by rdblue** [comment]
Looks good to me. Just one style issue to fix.

File: core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java (hunk 2 above)
Comment (shawnding) [reply core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java]: ok removed

**Comment by rdblue** [comment]
Merged. Thanks @shawnding!

**Pull request merged**